pkg/telemetry/events.go

```diff
@@ -145,6 +147,27 @@ func (t *telemetryService) ParticipantResumed(
 	reason livekit.ReconnectReason,
 ) {
 	t.enqueue(func() {
+		// create a worker if needed.
+		//
+		// Signalling channel stats collector and media channel stats collector could both call
+		// ParticipantJoined and ParticipantLeft.
+		//
+		// On a resume, the signalling channel collector would call `ParticipantLeft` which would close
+		// the corresponding participant's stats worker.
+		//
+		// So, on a successful resume, create the worker if needed.
+		_, found := t.getOrCreateWorker(
+			ctx,
+			livekit.RoomID(room.Sid),
+			livekit.RoomName(room.Name),
+			livekit.ParticipantID(participant.Sid),
+			livekit.ParticipantIdentity(participant.Identity),
+			true,
+		)
+		if !found {
+			prometheus.AddParticipant()
+		}
```

Comment:
if the participant was connected and hadn't left yet did we decrement the participant count? is this going to cause some resumed sessions to be double counted?

Reply:
Thought about that, I think it should be fine. But, I think I missed a bit here. Here is my thinking
- OSS case, signal connection close would call `ParticipantLeft` from the signalling collector. That would close the worker and reduce participant count. And when this creates a new worker, it will get bumped back up. Not perfect. Ideally would have been good to not decrement on the close, but did not want to introduce more complications in code.
- Cloud
  - Controller will close and create a new worker. So, it will bump down and up the count in controller prom.
  - Media will do `ParticipantResumed`, but it will be a no-op as a worker should be existing on media node.
  
  The part I missed is the new signalling collector not getting resolved. So, need to intercept `ReconnectResponse` and resolve room and participant. Will have a look.

Reply:
`ReconnectResponse` does not have `ParticipantInfo` understandably. So, getting it from `ParticipantInit` and resolving participant. Room resolve will happen on a room update. [Commit](https://github.com/livekit/livekit/pull/2982/commits/c68459c58998edd1d1b72e43fab9bf5d08bcb9ef).